### ISSUE TITLE
[BUG]: folia error

### ISSUE THREAD

**Issue by ewof** [SourceWriters/SmoothTimber#84]
### Describe what happend

```
[18:17:34 WARN]: [SmoothTimber] Async task for SmoothTimber v1.26.0 generated an exception
java.lang.NullPointerException: Cannot read field "captureTreeGeneration" because the return value of "net.minecraft.world.level.World.getCurrentWorldData()" is null
        at net.minecraft.world.level.Level.getBlockState(Level.java:1215) ~[?:?]
        at org.bukkit.craftbukkit.v1_20_R3.block.CraftBlock.getType(CraftBlock.java:238) ~[folia-1.20.4.jar:git-Folia-"25ee657"]
        at net.coreprotect.CoreProtectAPI.logRemoval(CoreProtectAPI.java:330) ~[CoreProtect-22.2.jar:?]
        at com.syntaxphoenix.spigot.smoothtimber.compatibility.coreprotect.CoreCompat_v1_13_x.logRemoval(CoreCompat_v1_13_x.java:19) ~[smoothtimber-legacy-1.26.0.jar:?]
        at com.syntaxphoenix.spigot.smoothtimber.compatibility.coreprotect.CoreProtectGrowListener.lambda$onGrow$0(CoreProtectGrowListener.java:29) ~[smoothtimber-legacy-1.26.0.jar:?]
        at com.syntaxphoenix.spigot.smoothtimber.platform.FoliaPlatform.lambda$asyncTaskLater$6(FoliaPlatform.java:82) ~[smoothtimber-legacy-1.26.0.jar:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaAsyncScheduler$AsyncScheduledTask.run(FoliaAsyncScheduler.java:217) ~[folia-1.20.4.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
        at java.lang.Thread.run(Thread.java:1583) ~[?:?]
```

### Reproduction steps

just run server

### Expected behavior

it doesnt happen

### Java Version

Java 18

### Server Software

folia

### Minecraft Version

1.20.4

### Plugin Version

1.26.0

### Additional Information

im on java 21

**Comment by Lauriichan** [comment]
I think this should not be reported to us but instead to CoreProtect or am I wrong about this?

**Comment by ewof** [comment]
if log removal is called on the location thread it doesnt happen

**Comment by Lauriichan** [comment]
> if log removal is called on the location thread it doesnt happen

Well yeah cause the issue here is that CoreProtect is trying to access the block of that location, therefore I would say it's an issue of CoreProtect which claims to support folia but if their api is not safe in that regard it doesn't seem like they fully support it.

Check here: https://github.com/PlayPro/CoreProtect/blob/master/src/main/java/net/coreprotect/CoreProtectAPI.java#L319

We can theoretically work around this bug of CoreProtect by scheduling on the regional thread ourselves yes however in my opinion this is as I stated already a work around and not a fix to the problem as the problem lies in the CoreProtect API which we are not responsible for

**Comment by ewof** [comment]
alr i made an issue in core protect

**Comment by Lauriichan** [comment]
Fixed in f6f3169fb9756401e2243981bee9c2d5ad35b01a after discussion with CoreProtect dev.
Their API has been expanded and in a later version regional access to the api will not be necessary
